File: app/src/main/java/com/akari/ppx/xp/hook/code/assist/RestrictHook.java
package com.akari.ppx.xp.hook.code.assist;

import com.akari.ppx.common.utils.XSP;
import com.akari.ppx.xp.hook.code.SuperbHook;

import de.robv.android.xposed.XC_MethodReplacement;

import static com.akari.ppx.common.constant.Prefs.REMOVE_RESTRICTIONS;
import static de.robv.android.xposed.XposedHelpers.getObjectField;

public class RestrictHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		if (!XSP.get(REMOVE_RESTRICTIONS)) return;
		hookMethod("com.sup.android.mi.feed.repo.bean.cell.AbsFeedItem", "isCanDownload", XC_MethodReplacement.returnConstant(true));
		hookMethod("com.sup.android.mi.feed.repo.bean.comment.Comment", "getCanDownload", XC_MethodReplacement.returnConstant(true));
		hookMethod("com.sup.android.mi.feed.repo.bean.cell.VideoFeedItem", "getVideoDownload", new XC_MethodReplacement() {
			@Override
			protected Object replaceHookedMethod(MethodHookParam param) {
				return getObjectField(param.thisObject, "originDownloadVideoModel");
			}
		});
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/assist/RestrictHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.assist

import com.akari.ppx.utils.getObjectField
import com.akari.ppx.utils.hookBeforeAllMethods
import com.akari.ppx.utils.replaceMethod
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.Init.webSharerClass
import com.akari.ppx.xp.hook.SwitchHook

class RestrictHook : SwitchHook("remove_download_restrictions") {
    override fun onHook() {
        "com.sup.android.mi.feed.repo.bean.cell.AbsFeedItem".replaceMethod(
            cl,
            "isCanDownload"
        ) { true }
        "com.sup.android.mi.feed.repo.bean.comment.Comment".replaceMethod(
            cl,
            "getCanDownload"
        ) { true }
        webSharerClass!!.hookBeforeAllMethods("showSharePanel") { param ->
            param.args[10] = true
        }
        "com.sup.android.mi.feed.repo.bean.cell.VideoFeedItem".replaceMethod(
            cl,
            "getVideoDownload"
        ) { param ->
            param.thisObject.getObjectField("originDownloadVideoModel")
        }
    }
}